Engine/CMakeLists.txt

```diff
@@ -171,102 +184,159 @@ set(VENDOR_PATH ${CMAKE_CURRENT_SOURCE_DIR}/vendor)
 # include dirs
 set(ENGINE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
-set(ENGINE_INCLUDE_DIRS
-${ENGINE_SOURCE_DIR}
-${VENDOR_PATH}/GLAD/include
-${VENDOR_PATH}/GLFW/include
-${VENDOR_PATH}/lia/include
-${VENDOR_PATH}/spdlog/include
-${VENDOR_PATH}/stb/include
-${VENDOR_PATH}/json/include
-${VENDOR_PATH}/freetype/include
-${VENDOR_PATH}/fmt/include
+# ------------------
+# Dependencies
+# ------------------
+FetchContent_Declare(
+        glfw
+        GIT_REPOSITORY "https://github.com/glfw/glfw.git"
+        GIT_TAG "3.4"
+)
+FetchContent_MakeAvailable(glfw)
+
+FetchContent_Declare(
+        lia
+        GIT_REPOSITORY "https://github.com/denyskryvytskyi/lia.git"
+        GIT_TAG "main"
+)
+FetchContent_MakeAvailable(lia)
+
+FetchContent_Declare(
+        spdlog
+        GIT_REPOSITORY "https://github.com/gabime/spdlog.git"
+        GIT_TAG "v1.14.1"
+)
+FetchContent_MakeAvailable(spdlog)
+
+FetchContent_Declare(json URL "https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz")
+FetchContent_MakeAvailable(json)
+
+FetchContent_Declare(
+        freetype
+        GIT_REPOSITORY "https://github.com/freetype/freetype.git"
+        GIT_TAG "VER-2-13-2"
+)
+FetchContent_MakeAvailable(freetype)
+
+FetchContent_Declare(
+        fmt
+        GIT_REPOSITORY "https://github.com/fmtlib/fmt.git"
+        GIT_TAG "10.2.1"
+)
+FetchContent_MakeAvailable(fmt)
+
+FetchContent_Declare(assimp
+GIT_REPOSITORY "https://github.com/assimp/assimp.git"
+GIT_TAG "v5.3.1")
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
+set(ASSIMP_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+set(ASSIMP_INJECT_DEBUG_POSTFIX OFF CACHE BOOL "" FORCE)
+set(ASSIMP_INSTALL OFF CACHE BOOL "" FORCE)
+set(ASSIMP_BUILD_ALL_EXPORTERS_BY_DEFAULT OFF CACHE BOOL "" FORCE)
+set(ASSIMP_BUILD_ZLIB OFF CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(assimp)
+
+FetchContent_Declare(
+  imgui
+  GIT_REPOSITORY https://github.com/ocornut/imgui
+  GIT_TAG "docking"
+)
+
+FetchContent_MakeAvailable(imgui)
+
+
+FetchContent_Declare(
+        zlib
+        GIT_REPOSITORY "https://github.com/madler/zlib.git"
+        GIT_TAG "v1.3.1"
+)
+FetchContent_MakeAvailable(zlib)
+
+
+set(IRRKLANG_INSTALL_DIR ${CMAKE_BINARY_DIR}/_deps/irrklang)
+
+FetchContent_Declare(
+    irrklang
+	URL "http://www.ambiera.at/downloads/irrKlang-64bit-1.5.0.zip"
+    SOURCE_DIR ${IRRKLANG_INSTALL_DIR}
 )
 
+FetchContent_GetProperties(irrklang)
+
+if(NOT irrklang_POPULATED)
+	FetchContent_Populate(irrklang)
+	set(IRRKLANG_BUILD_DOCS OFF CACHE BOOL "" FORCE)
+	set(IRRKLANG_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+	set(IRRKLANG_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+endif()
+
+add_subdirectory(${VENDOR_PATH}/GLAD)
+add_subdirectory(${VENDOR_PATH}/stb)
```

Comment:
GLAD and stb are the only libraries that are not FetchContent'd, as they are small enough and filtering out the stb_image from whole repo of stb is a bit of a hassle.